src/components/WithdrawalCredentials.tsx

```diff
@@ -0,0 +1,167 @@
+  const [validator, setValidator] = useState<Validator | null>(null);
+  const isMobile: boolean = (window as any).mobileCheck();
+
+  const checkWithdrawalCredentials = async () => {
+    setHasError(false);
+    setIsLoading(true);
+    const endpoint = `${BEACONCHAIN_URL}/api/v1/validator/${inputValue}`;
+    try {
+      const response = await fetch(endpoint);
+      const { data } = await response.json();
+      const withdrawalCredentials = data.withdrawalcredentials;
+      setValidator({
+        validatorIndex: parseInt(inputValue, 10),
+        withdrawalCredentials,
+        isUpgraded: withdrawalCredentials.startsWith('0x01'),
+      });
+    } catch (error) {
+      setHasError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) =>
+    setInputValue(e.target.value.replace(/\D/g, ''));
+
+  const longAddress = useMemo<string>(
+    () => (validator ? `0x${validator.withdrawalCredentials.slice(-40)}` : ''),
+    [validator]
+  );
+  const shortAddress = useMemo<string>(
+    () =>
+      longAddress ? `${longAddress.slice(0, 6)}…${longAddress.slice(-4)}` : '',
+    [longAddress]
+  );
+  const resultText = useMemo<string | JSX.Element>(() => {
+    if (isLoading) return <Spinner />;
+    if (hasError)
+      return (
+        <Alert variant="error">
+          <FormattedMessage defaultMessage="Oops! Double check validator index number and try again." />
+        </Alert>
+      );
+    if (!validator) return ' ';
+    if (validator.isUpgraded)
+      return (
+        <Alert variant="success">
+          <FormattedMessage
+            defaultMessage="Validator index {validatorIndex} is ready to start
+            receiving rewards! Withdrawal credentials linked to execution address {address}"
+            values={{
+              validatorIndex: (
+                <strong>
+                  {new Intl.NumberFormat(locale).format(
+                    validator.validatorIndex
+                  )}
+                </strong>
+              ),
+              address: <strong title={longAddress}>{shortAddress}</strong>,
+            }}
+          />
+        </Alert>
+      );
+    return (
+      <Alert variant="error">
+        <FormattedMessage
+          defaultMessage="This {network} validator needs to be upgraded."
+          values={{
+            network: IS_MAINNET ? (
+              ''
+            ) : (
+              <FormattedMessage
+                defaultMessage="{NETWORK_NAME} testnet"
+                values={{ NETWORK_NAME }}
+              />
+            ),
+          }}
+        />
+      </Alert>
+    );
+  }, [isLoading, hasError, validator, longAddress, shortAddress, locale]);
+
+  return (
+    <Container>
+      <FlexRow>
+        <Input
+          id="validatorIndex"
+          value={inputValue}
+          onChange={handleChange}
+          placeholder="Validator index"
+        />
+        <Button
```

Comment:
@konopkja ^ I had matched your design suggestions, but to Carl's point, this theming is not implemented anywhere else on the site. Would personally suggest we match existing styles. If we want to do larger design changes to the site that could be discussed, but I'd try to avoid too many individual deviations before a holistic approach is taken.

Reply:
ok agree